### PR TITLE
Perf optimizations

### PR DESCRIPTION
**Pull request by divmain**
This PR includes some performance improvements.  We will also be dropping Radium from Victory, and providing docs for how to use the two together.  But that change will occur in a separate PR.

Here's what performance looks like after the changes (including removal of Radium).  This is a ~3-4x improvement over the original performance.

relative-performance

/cc @boygirl @coopy

### CODE REVIEW

File: src/helpers/domain.js [hunk 1]
```diff
@@ -73,8 +73,8 @@ module.exports = {
       return this.getDomainFromCategories(props, axis);
     }
     // find the global min and max
-    const rawDatasets = (props.stacked || this.shouldGroup(props)) ? props.data : [props.data];
-    const datasets = Data.formatDatasets(rawDatasets, props)
+    const hasMultipleDatasets = props.stacked || this.shouldGroup(pro
```
Comment (exogen) [inline src/helpers/domain.js]: Line cut off?

Comment (divmain) [inline src/helpers/domain.js]: Weird...

**Comment by coopy** [comment]
This might work for statically styled components (I'll let @boygirl comment on the new API); we still need to address perf in cases where styles are dynamic.

File: src/components/victory-bar/victory-bar.jsx [hunk 2]
```diff
@@ -251,6 +252,12 @@ export default class VictoryBar extends React.Component {
 
   static getDomain = BarHelpers.getDomain.bind(BarHelpers);
 
+  componentWillMount() {
+    this.memoized = {
+      getStyles: memoizerific(1)(Helpers.getStyles)
```
Comment (ryan-roemer) [inline src/components/victory-bar/victory-bar.jsx]: Can we get a code comment as to what `memoizerific(1)` is doing?

**Comment by divmain** [comment]
Hmm... tests are passing locally but not in CI.  I'm investigating.

File: src/components/victory-axis/tick.jsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React, { PropTypes } from "react";
 import Radium from "radium";
-import { VictoryLabel } from "victory-label";
-import { Helpers } from "victory-util";
+import { VictoryLabel } from "victory-core";
+import { Helpers } from "victory-core";
```
Comment (coopy) [inline src/components/victory-axis/tick.jsx]: `import { Helpers, VictoryLabel } from "victory-core";`

File: src/components/victory-axis/victory-axis.jsx [hunk 4]
```diff
@@ -2,13 +2,13 @@ import merge from "lodash/object/merge";
...
-import { VictoryLabel } from "victory-label";
-import { VictoryAnimation } from "victory-animation";
+import { VictoryLabel } from "victory-core";
+import { VictoryAnimation } from "victory-core";
 import AxisLine from "./axis-line";
 import GridLine from "./grid";
 import Tick from "./tick";
 import AxisHelpers from "./helper-methods";
-import { PropTypes as CustomPropTypes, Helpers } from "victory-util";
+import { PropTypes as CustomPropTypes, Helpers } from "victory-core";
```
Comment (coopy) [inline src/components/victory-axis/victory-axis.jsx]:
```
import {
  PropTypes as CustomPropTypes,
  Helpers,
  VictoryAnimation,
  VictoryLabel
} from "victory-core";
```

File: src/components/victory-bar/bar-label.jsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import merge from "lodash/object/merge";
 import React, { PropTypes } from "react";
 import Radium from "radium";
-import { VictoryLabel } from "victory-label";
-import { Helpers } from "victory-util";
+import { VictoryLabel } from "victory-core";
+import { Helpers } from "victory-core";
```
Comment (coopy) [inline src/components/victory-bar/bar-label.jsx]: same

File: src/components/victory-line/line-label.jsx [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import merge from "lodash/object/merge";
 import React, { PropTypes } from "react";
 import Radium from "radium";
-import {VictoryLabel} from "victory-label";
-import { Helpers } from "victory-util";
+import {VictoryLabel} from "victory-core";
+import { Helpers } from "victory-core";
```
Comment (coopy) [inline src/components/victory-line/line-label.jsx]: same

File: src/components/victory-bar/bar-label.jsx (hunk 5 above)
Comment (divmain) [inline src/components/victory-bar/bar-label.jsx]: agh, good catch.

File: src/components/victory-line/victory-line.jsx [hunk 7]
```diff
@@ -13,8 +13,9 @@ import LineLabel from "./line-label";
 import Scale from "../../helpers/scale";
 import Domain from "../../helpers/domain";
 import Data from "../../helpers/data";
-import { PropTypes as CustomPropTypes, Helpers } from "victory-util";
-import { VictoryAnimation } from "victory-animation";
+import { PropTypes as CustomPropTypes, Helpers } from "victory-core";
+import { VictoryAnimation } from "victory-core";
```
Comment (coopy) [inline src/components/victory-line/victory-line.jsx]: ^

File: src/components/victory-scatter/point.jsx [hunk 8]
```diff
@@ -3,9 +3,9 @@ import omit from "lodash/object/omit";
 import pick from "lodash/object/pick";
 import React, { PropTypes } from "react";
 import Radium from "radium";
-import {VictoryLabel} from "victory-label";
+import {VictoryLabel} from "victory-core";
 import { getPath } from "./helper-methods";
-import { Helpers } from "victory-util";
+import { Helpers } from "victory-core";
```
Comment (coopy) [inline src/components/victory-scatter/point.jsx]: ^

File: src/components/victory-scatter/victory-scatter.jsx [hunk 9]
```diff
@@ -5,9 +5,10 @@ import Point from "./point";
 import Scale from "../../helpers/scale";
 import Domain from "../../helpers/domain";
 import Data from "../../helpers/data";
-import { PropTypes as CustomPropTypes, Helpers } from "victory-util";
-import { VictoryAnimation } from "victory-animation";
+import { PropTypes as CustomPropTypes, Helpers } from "victory-core";
+import { VictoryAnimation } from "victory-core";
```
Comment (coopy) [inline src/components/victory-scatter/victory-scatter.jsx]: ^

File: src/helpers/data.js [hunk 10]
```diff
@@ -105,31 +105,37 @@ export default {
       .map((datum) => {
         const x = accessor.x(datum);
         const y = accessor.y(datum);
-        return merge({}, datum, {
-          category: this.determineCategoryIndex(x, props.categories),
+        const category = this.determineCategoryIndex(x, props.categories);
+
+        return assign(
```
Comment (coopy) [inline src/helpers/data.js]: We should be able to just use `Object.assign` and drop the lodash `assign` module.

**Comment by coopy** [comment]
Looks good - just some style / trivial comments. PERF++!

File: src/helpers/data.js (hunk 10 above)
Comment (divmain) [inline src/helpers/data.js]: Seeing `TypeError: 'undefined' is not a function (evaluating 'Object.assign')`.  Investigating whether `Object.assign` support is included in `builder-victory-component`.

**Comment by boygirl** [comment]
@divmain I think there are still a few references to victory-label in the demos

**Comment by boygirl** [comment]
hmm, actually all of the demos are broken except for the axis demo. Now I'm sad that CI is passing :( Are you able to reproduce this problem @divmain

**Comment by divmain** [comment]
@boygirl I didn't think to check the demos.  Figured `builder run check` was adequate.  I'll do that now!

**Comment by divmain** [comment]
Alright, this should be good to go.  Still passing `builder run check` locally, and the demo is working as expected.  Ping for final review @boygirl and @coopy.

**Comment by boygirl** [comment]
cc/ @divmain some of the demos are still importing VictoryLabel from `victory-label`. It's probably still hanging around in your `node_modules`. Please change these to depend on `victory-core`. Offenders are `demos/victory-axis-demo`, `demos/victory-line-demo`, `demos/victory-scatter-demo` and `docs/victory-bar/docs.jsx`. Approved once these are resolved.

**Comment by divmain** [comment]
Meh.  Sorry about that.

**Comment by divmain** [comment]
@boygirl is there a way to check health of demos from CI?

**Comment by boygirl** [comment]
Currently no. Since they are all their own components though, we could at least be trying to render them in tests. What I would ideally like would be screenshot comparison, but there still isn't a super good solution for that as far as I can tell. I'll open an issue for the easy win though.

**Comment by boygirl** [comment]
https://github.com/FormidableLabs/victory-chart/issues/110

**Comment by divmain** [comment]
thxmuch :)

**Pull request merged**